pilot/pkg/proxy/envoy/watcher.go

```diff
@@ -116,6 +116,9 @@ func (w *watcher) Reload() {
 // retrieveAZ will only run once and then exit because AZ won't change over a proxy's lifecycle
 // it has to use a reload due to limitations with envoy (az has to be passed in as a flag)
 func (w *watcher) retrieveAZ(ctx context.Context, delay time.Duration, retries int) {
+	if w.config.AvailabilityZone == "" {
```

Comment:
Well, it is a good change - but still a bit miss-leading, if the next code connects to pilot it will still fetch an AZ ? LGTM otherwise.

We may need to revisit this code later, we may want to fetch more than AZ from pilot, and even maybe cache the result - so I guess quite a few PRs may adjust this.